src/modules/ethereum/exchangeTransactions.js

```diff
@@ -211,6 +182,111 @@ export async function matchOrdersTx(buyId, sellId) {
     return { txName, transactionHash };
 }
 
+export async function matchMultipleOrdersTx() {
+    const txName = "Match orders";
+    const userAccount = store.getState().web3Connect.userAccount;
+    const exchange = store.getState().contracts.latest.exchange.web3ContractInstance;
+    const bn_ethFiatRate = store.getState().rates.info.bn_ethFiatRate;
+    const orders = store.getState().orders.orders;
+    const lowestSellPrice = orders.sellOrders[0].price;
+    const highestBuyPrice = orders.buyOrders[0].price;
+
+    const buyOrders = orders.buyOrders
+        .filter(o => o.price >= lowestSellPrice)
+        .map(o => ({ id: o.id, price: o.price, bn_ethAmount: o.bn_ethAmount }));
+    const sellOrders = orders.sellOrders
+        .filter(o => o.price <= highestBuyPrice)
+        .map(o => ({ id: o.id, price: o.price, bn_tokenAmount: new BigNumber(o.amount) }));
+
+    const sellIds = [];
+    const buyIds = [];
+    let buyIdx = 0;
+    let sellIdx = 0;
+    let matchCount = 0;
+    while (
+        buyIdx < buyOrders.length &&
+        sellIdx < sellOrders.length &&
+        buyOrders[buyIdx].price >= sellOrders[sellIdx].price && // we already filtered but just in case...
```

Comment:
if the filter logic above was correct, remove this. or remove the filtering above :)